Userland/Libraries/LibWeb/Loader/FileDirectoryLoader.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023, Bastiaan van der Plaat <bastiaan.v.d.plaat@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/QuickSort.h>
+#include <AK/SourceGenerator.h>
+#include <LibCore/DateTime.h>
+#include <LibCore/Directory.h>
+#include <LibCore/System.h>
+#include <LibWeb/Loader/FileDirectoryLoader.h>
+#include <LibWeb/Loader/FrameLoader.h>
+
+namespace Web {
+
+FileDirectoryLoader& FileDirectoryLoader::the()
+{
+    static FileDirectoryLoader instance;
+    return instance;
+}
+
+ErrorOr<DeprecatedString> FileDirectoryLoader::load(LoadRequest const& request) const
+{
+    // Generate HTML contents entries table
+    auto lexical_path = LexicalPath(request.url().serialize_path());
+    Core::DirIterator dt(lexical_path.string(), Core::DirIterator::Flags::SkipParentAndBaseDir);
+    Vector<DeprecatedString> names;
+    while (dt.has_next())
+        names.append(dt.next_path());
+    quick_sort(names);
+
+    StringBuilder contents;
+    contents.append("<table>"sv);
```

Comment:
How does this structure compare with what you see when loading a directory from `Services/WebServer`? It would be good to see if they can be unified. Maybe they should share the code?

Reply:
It is almost the same, ideally the code would be unified. But I don't no how to do that cleanly. A random dependency from `LibWeb` to `Services/WebServer` or back would be strange. And another `LibX` is overkill. If you have any better idea let me now! But for now I think this makes the most sense.

Reply:
But for now I think that my new class `FileDirectoryLoader.cpp` will do